src/main/notebooks/TPC-multi_datagen.scala

```diff
@@ -0,0 +1,277 @@
+val createTableStats = true
+val createColumnStats = true
+
+val workers: Int = spark.conf.get("spark.databricks.clusterUsageTags.clusterTargetWorkers").toInt //number of nodes, assumes one executor per node
+val cores: Int = Runtime.getRuntime.availableProcessors.toInt //number of CPU-cores
+
+val dbSuffix = "" // set only if creating multiple DBs or source file folders with different settings, use a leading _
+val TPCDSUseLegacyOptions = false // set to generate file/DB naming and table options compatible with older results
+
+// COMMAND ----------
+
+// Imports, fail fast if we are missing any library
+
+// For datagens
+import java.io._
+import scala.sys.process._
+
+// spark-sql-perf
+import com.databricks.spark.sql.perf._
+import com.databricks.spark.sql.perf.tpch._
+import com.databricks.spark.sql.perf.tpcds._
+
+// Spark/Hadoop config
+import org.apache.spark.deploy.SparkHadoopUtil
+
+// COMMAND ----------
+
+// Set Spark config to produce same and comparable source files across systems 
+// do not change unless you want to derive from default source file composition, in that case also set a DB suffix 
+spark.sqlContext.setConf("spark.sql.parquet.compression.codec", "snappy")
+
+// Prevent very large files. 20 million records creates between 500 and 1500MB files in TPCH
+spark.sqlContext.setConf("spark.sql.files.maxRecordsPerFile", "20000000")  // This also regulates the file coalesce
+
+// COMMAND ----------
+
+// Checks that we have the correct number of worker nodes to start the data generation
+// Make sure you have set the workers variable correctly, as the datagens binaries need to be present in all nodes
+val targetWorkers: Int = workers
+def numWorkers: Int = sc.getExecutorMemoryStatus.size - 1
+def waitForWorkers(requiredWorkers: Int, tries: Int) : Unit = {
+  for (i <- 0 to (tries-1)) {
+    if (numWorkers == requiredWorkers) {
+      println(s"Waited ${i}s. for $numWorkers workers to be ready")
+      return
+    }
+    if (i % 60 == 0) println(s"Waiting ${i}s. for workers to be ready, got only $numWorkers workers")
+    Thread sleep 1000
+  }
+  throw new Exception(s"Timed out waiting for workers to be ready after ${tries}s.")
+}
+waitForWorkers(targetWorkers, 3600) //wait up to an hour
+
+// COMMAND ----------
+
+// Time command helper
+def time[R](block: => R): R = {  
+    val t0 = System.currentTimeMillis() //nanoTime()
+    val result = block    // call-by-name
+    val t1 = System.currentTimeMillis() //nanoTime()
+    println("Elapsed time: " + (t1 - t0) + "ms")
+    result
+}
+
+// COMMAND ----------
+
+// FOR INSTALLING TPCH DBGEN (with the sdtout patch)
+def installDBGEN(url: String = "https://github.com/databricks/tpch-dbgen.git", useStdout: Boolean = true, baseFolder: String = "/tmp")(i: java.lang.Long): String = {
+  // check if we want the revision which makes dbgen output to stdout
+  val checkoutRevision: String = if (useStdout) "git checkout 0469309147b42abac8857fa61b4cf69a6d3128a8 -- bm_utils.c" else ""
```

Comment:
would we want not to do that?
we always do this tweak with stdout for dsdgen, so maybe this option here adds complexity, and adds a well buried dependency on a concrete githash in the repo.

Reply:
We might want to keep the source csv files or test the full import from file system. I prefer to have this option available.